.github/scripts/attack.py

```
@@ -336,8 +336,72 @@ def attack_test():
     time.sleep(DURATION_SEC)
     stop.set()
 
-    # Teardown
-    for t in pubs:
-        t.join(timeout=2)
+    # Teardown: wait for every worker, including churn/flapper, so no client
+    # keeps reconnecting as X while we clean up below
+    for t in pubs + [t_churn, t_flap]:
```

Ruff (0.15.21)

[warning] Consider `[*pubs, t_churn, t_flap]` instead of concatenation

Replace with `[*pubs, t_churn, t_flap]`

(RUF005)